floodlight/io/dfl.py

```diff
@@ -342,24 +281,120 @@ def read_pitch_from_mat_info_xml(filepath_mat_info: Union[str, Path]) -> Pitch:
+        ]
+        teamsheets[team]["pID"] = [
+            player.get("PersonId") for player in team_info.find("Players")
+        ]
+        teamsheets[team]["jID"] = [
+            int(player.get("ShirtNumber")) for player in team_info.find("Players")
+        ]
+        teamsheets[team]["position"] = [
+            player.get("PlayingPosition") for player in team_info.find("Players")
+        ]
+        teamsheets[team]["tID"] = team_info.get("TeamId")
+        teamsheets[team]["team"] = team_info.get("TeamName")
+
+    # create teamsheet objects
+    for team in teamsheets:
+        teamsheets[team] = Teamsheet(teamsheets[team])
+
+    return teamsheets
+
+
+def read_event_data_xml(
+    filepath_events: Union[str, Path],
+    filepath_mat_info: Union[str, Path],
+    home_teamsheet: Teamsheet = None,
+    away_teamsheet: Teamsheet = None,
+) -> Tuple[Events, Events, Events, Events, Teamsheet, Teamsheet]:
+    """Parses a DFL Match Event XML file and extracts the event data as well as
+    teamsheets.
+
+    The structure of the official tracking system of the DFL (German Football League)
+    contains two separate xml files, one containing the actual data as well as a
+    metadata file containing information about teams, pitch size, and start- and
+    endframes of match periods. This function provides high-level access to DFL data by
+    parsing "the full match" and returning Events-objects parsed from the event data
+    xml-file as well as Teamsheet-objects parsed from the metadata xml-file. The number
+    of segments is inferred from the data, yet data for each segment is stored in a
+    separate object.
 
     Parameters
     ----------
     filepath_events: str or pathlib.Path
         Full path to XML File where the Event data in DFL format is saved.
+    filepath_mat_info: str or pathlib.Path
+        Full path to XML File where the Match Information data in DFL format is saved.
+    home_teamsheet: Teamsheet, optional
+        Teamsheet-object for the home team used to create link dictionaries of the form
+        `links[team][jID] = xID` and  `links[team][pID] = jID`. The links are used to
+        map players to a specific xID in the respective XY objects. Should be supplied
+        if that order matters. If given as None (default), teamsheet is extracted from
+        the Match Information XML file.
+    away_teamsheet: Teamsheet, optional
+        Teamsheet-object for the away team. If given as None (default), teamsheet is
+        extracted from the Match Information XML file.
 
     Returns
     -------
-    data_objects: Tuple[Events, Events, Events, Events]
-        Events- and Pitch-objects for both teams and both halves.
+    data_objects: Tuple[Events, Events, Events, Events, Teamsheet, Teamsheet]
+        Events- and Pitch-objects for both teams and both halves. The order is
+        (events_home_ht1, events_home_ht2, events_away_ht1, events_away_ht2,
+        home_teamsheet, away_teamsheet).
```

Comment:
Doesn't return pitch objects, right? Although that might actually be a good idea. In case someone (certainly a maniac) is using DFL event data with xy data from a different provider

Reply:
Agreed, since MatchInformation XML needs to be passed for the teamsheets, anyway! 

However, regarding the other parsers this only makes sense if pitch information is stored in a seperate file (like here) or if pitch information is contained in the data (events or position) file itself. Saying that, this will lead to some parsers returning Pitch objects while some parsers do not.

Reply:
Which I believe is ok, btw!

Reply:
Agreed!